TowerDefense/MouseHandler.cs

```diff
@@ -106,6 +102,14 @@ public void Update(GameEngine gameEngine, GameMap gameMap)
 
             HoveringContext = HoveringContext.None;
             HoveredObject = null;
+
+            base.Update(gameTime);
+        }
+
+        public override void Draw(SpriteBatch batch)
+        {
+
+            batch.Draw(Tex, Position.ToVector2(), new Rectangle(new Point(currentFrame * spriteWidth, 0), new Point(spriteWidth, spriteHeight)), Color, 0, Vector2.Zero, 1, SpriteEffects.None, 0);
```

Comment:
The base draw should handle animation, but it doesn't mostly because of nodes possibly having 2 textures, and a few other edge cases where stuff broke when I did that.  I'll try to refactor that to be possible soon.